h5py/h5d.pyx

```diff
@@ -405,7 +409,7 @@ cdef class DatasetID(ObjectID):
 
     IF HDF5_VERSION >= (1, 8, 11):
 
-        def write_direct_chunk(self, offsets, bytes data, filter_mask=0x00000000, PropID dxpl=None):
+        def write_direct_chunk(self, offsets, data, filter_mask=0x00000000, PropID dxpl=None):
```

Comment:
we probably shouldn't change the API  like this

Reply:
But this only relaxes the type constraint and it doesn't change how the function is called. So it should be transparent to anyone using the API and calling it with bytes still works without any change. It just allows other types as well

Reply:
But it removes a positional argument, which causes all downstream calls of this function to break :cry:

Reply:
@scopatz I think you're misreading the change - it's not removing an argument, just removing the `bytes` type from the `data` argument (a bit of C like Cython syntax).

Reply:
Oh! I am! Sorry.  Serves me right for trying to reveiw on my phone